### DISCUSSION TITLE
为什么引入依赖包要涉及到两个接口

### DISCUSSION THREAD

**Discussion by tao-hengyi** [xmake-io/xmake#4271] [Ideas]
### Is your feature request related to a problem? Please describe.

引入一个依赖包需要先使用`add_require()`，然后再使用`add_package()`，语义上有点重复。

### Describe the solution you'd like

取消最开始的`add_require()`，只在每个target下使用`add_package()`

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
不重复，功能不同。。

1. add_requires 是全局接口，配置需要安装的包，已经每个包的具体特性、版本、static/shared, debug/release 等配置。
配置了这个接口，仅用于告诉 xmake ，当前怎么安装包，不代表 target 就会去用它。

它类似于 cargo.toml 的配置

2. add_packages 是 target 粒度接口，会根据 add_requires 的包配置，对特定 target 关联绑定 links, includedirs 等实际编译链接所需要的信息。它是不包含包的配置信息的。

这个类似于 add_links/add_includedirs/add_frameworks/add_ldflags 的集合

完全不一样的东西，你要合并成一个接口，很多特性就要受到限制，并且每个 target 都要完整配置一长串的 包配置 ，很繁琐。。

[a comment by an automated account is not shown]
